source/Applications/Advanced/MaskPointCloud/MaskPointCloud.cpp

```diff
@@ -0,0 +1,222 @@
+/*
+This example shows how to mask a point cloud from ZDF file and store
+the resulting point cloud in PCL format. The ZDF file for this sample can be found under the
+main instructions for Zivid samples.
+*/
+
+#include <Zivid/Zivid.h>
+
+#include <opencv2/opencv.hpp>
+
+#include <pcl/visualization/cloud_viewer.h>
+
+#include <iostream>
+
+namespace
+{
+    void visualizePointCloud(const pcl::PointCloud<pcl::PointXYZRGB> &pointCloud)
+    {
+        pcl::PointCloud<pcl::PointXYZRGB>::Ptr cloudPTR(new pcl::PointCloud<pcl::PointXYZRGB>);
+        *cloudPTR = pointCloud;
+        std::cout << "Running PCL visualizer. Block until window closes" << std::endl;
+        pcl::visualization::CloudViewer viewer("Simple Cloud Viewer");
+        viewer.showCloud(cloudPTR);
+        std::cout << "Press r to centre and zoom the viewer so that the entire cloud is visible" << std::endl;
+        std::cout << "Press q to me exit the viewer application" << std::endl;
+        while(!viewer.wasStopped())
+        {
+        }
+    }
+
+    cv::Mat pointCloudToCvZ(const pcl::PointCloud<pcl::PointXYZRGB> &pointCloud)
+    {
+        // Getting min and max values for X, Y, Z images
+        float zMax = -1;
+        float zMin = 1000000;
+        for(size_t i = 0; i < pointCloud.height; i++)
+        {
+            for(size_t j = 0; j < pointCloud.width; j++)
+            {
+                if(pointCloud(j, i).z > zMax)
+                {
+                    zMax = pointCloud(j, i).z;
+                }
+                if(pointCloud(j, i).z < zMin)
+                {
+                    zMin = pointCloud(j, i).z;
+                }
+            }
+        }
+
+        // Filling in OpenCV matrix with the cloud data
+        cv::Mat z(pointCloud.height, pointCloud.width, CV_8UC1, cv::Scalar(0)); // NOLINT(hicpp-signed-bitwise)
+        for(size_t i = 0; i < pointCloud.height; i++)
+        {
+            for(size_t j = 0; j < pointCloud.width; j++)
+            {
+                if(std::isnan(pointCloud(j, i).z))
+                {
+                    z.at<uchar>(i, j) = 0;
+                }
+                else
+                {
+                    z.at<uchar>(i, j) =
+                        static_cast<unsigned char>((255.0F * (pointCloud(j, i).z - zMin) / (zMax - zMin)));
+                }
+            }
+        }
+
+        // Applying color map
+        cv::Mat zJetColorMap;
+        cv::applyColorMap(z, zJetColorMap, cv::COLORMAP_JET);
+
+        // Setting invalid points (nan) to black
+        for(size_t i = 0; i < pointCloud.height; i++)
+        {
+            for(size_t j = 0; j < pointCloud.width; j++)
+            {
+                if(std::isnan(pointCloud(j, i).z))
+                {
+                    auto &zRGB = zJetColorMap.at<cv::Vec3b>(i, j);
+                    zRGB[0] = 0;
+                    zRGB[1] = 0;
+                    zRGB[2] = 0;
+                }
+            }
+        }
+        return zJetColorMap;
```

Comment:
Check this out: https://github.com/zivid/zivid-cpp-samples/blob/d840629c50b18d6428ae6749774497f310980dbf/source/Applications/Advanced/CreateDepthMap/CreateDepthMap.cpp#L57-L83

It does the same thing but it is more optimized, e.g. `pointCloud.copyPointsZ();` is much faster than doing this in a loop, and it also shows how to utilize our API with regards to OpenCV, finding min and max is just two lines of code so it looks nicer, etc.

Reply:
Note that we have decided to use COLORMAP_VIRIDIS as default.